CMake/ITKSetPython3Vars.cmake

```diff
@@ -36,18 +37,16 @@ if("${CMAKE_VERSION}" VERSION_LESS_EQUAL "3.12.0")
     set(Python3_LIBRARIES    ${PYTHON_LIBRARIES})
   endif()
 else()
+  cmake_policy(SET CMP0094 NEW) # makes FindPython3 prefer activate virtualenv Python to latest version
   if(PYTHON_DEVELOPMENT_REQUIRED)
-    if(DEFINED Python3_EXECUTABLE)
-      set(_specified_Python3_EXECUTABLE ${Python3_EXECUTABLE})
-    endif()
```

Comment:
We need to keep this around. FindPython3 will overwrite a `Python3_EXECUTABLE` specified with `-D` :cry: :disappointed: *silently* :sob: :disappointed: **with the wrong value** :crying_cat_face: :disappointed: .

Reply:
@thewtex Is ```_specified_Python3_EXECUTABLE``` used or specified anywhere?  With the underscore, it looks like an internal variable.  But, when I grep for it, I don't find it referenced or declared anywhere but here.  The way the original code was written, if that was unspecified, it would **always** overwrite the ```Python3_EXECUTABLE```.

Reply:
@thewtex 
``` 
 if(PYTHON_DEVELOPMENT_REQUIRED)
    if(DEFINED Python3_EXECUTABLE) <-- why would this be defined, we haven't called FindPython
      set(_specified_Python3_EXECUTABLE ${Python3_EXECUTABLE})
    endif()
    find_package(Python3 COMPONENTS Interpreter Development)
    set(Python3_EXECUTABLE ${_specified_Python3_EXECUTABLE} CACHE INTERNAL
      "Path to the Python interpreter" FORCE) <-- We force Python3_EXECUTABLE to be our undefine variable
  else()
    find_package(Python3 COMPONENTS Interpreter)
  endif()
  if(NOT Python3_EXECUTABLE AND _Python3_EXECUTABLE) <--- NOT Python3_EXECUTABLE is true if _specified is undefined because we overwrite it earlier
    set(Python3_EXECUTABLE ${_Python3_EXECUTABLE} CACHE INTERNAL
      "Path to the Python interpreter" FORCE) <-- _Python3_EXECUTABLE is an internal CMake variable, and is likely defined by our call to FindPython3, but why are we referencing it here?
  endif()
endif()
```

Reply:
I think with the latest find package you have to specify ```Python3_ROOT_DIR``` and not ```Python3_EXECUTABLE```.

Reply:
> if(DEFINED Python3_EXECUTABLE) <-- why would this be defined, we haven't called FindPython

This is defined if a user specifies it as a CMake configuration variable, stating -- "this is the Python executable I want to use."